src/nvim/api/vim.c

```diff
@@ -2451,3 +2452,32 @@ void nvim_del_user_command(String name, Error *err)
 {
   nvim_buf_del_user_command(-1, name, err);
 }
+
+/// Get the command line text.
+///
+/// @param[out] err Error details, if any.
+String nvim_get_cmdline(Error *err)
+    FUNC_API_SINCE(9)
+{
+    String res = STRING_INIT;
+
+    char *cmdline;
+    if (!(cmdline = (char *)get_cmdline_str())) {
+        api_set_error(err, kErrorTypeException, "Can't set command line when not active or entering a password");
+        return res;
+    }
+
+    return cstr_to_string(cmdline);
+}
+
+/// Set the command line text to the given string.
+///
+/// @param  cmdline String to set the command line as.
+/// @param[out] err Error details, if any.
+void nvim_set_cmdline(String name, Error *err)
```

Comment:
The argument name matters because it's what shows up in the docs.

```suggestion
void nvim_set_cmdline(String cmdline, Error *err)
```